core/app/models/mno_enterprise/tenant_config.rb

```diff
@@ -121,6 +113,39 @@ def self.build_object(schema)
       end
     end
 
+    # Flag field as readonly in the config schema
+    # A field is readonly if the `<field>_readonly` key is `true` in the `frontend_config`
+    def self.update_readonly_fields(schema, frontend_config)
+      frontend_config.each do |current_key, value|
+        if current_key =~ /_readonly$/
```

Comment:
I would suggest `current_key.end_with?("_readonly")`

Reply:
Or you can do this 😎 
```ruby
if field_label = current_key[/(.*)_readonly$/, 1]
  schema['properties'][field_label]['readonly'] = true
end
```

PS: rubocop recommends assignments rather than merge